usaspending_api/download/management/commands/generate_spark_download.py

```diff
@@ -0,0 +1,224 @@
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils.functional import cached_property
+from pyspark.sql import SparkSession
+
+from usaspending_api.common.etl.spark import create_ref_temp_views
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.common.helpers.download_csv_strategies import SparkToCSVStrategy
+from usaspending_api.common.helpers.s3_helpers import upload_download_file_to_s3
+from usaspending_api.common.helpers.spark_helpers import (
+    configure_spark_session,
+    get_active_spark_session,
+    get_jdbc_connection_properties,
+    get_jvm_logger,
+    get_usas_jdbc_url,
+)
+from usaspending_api.download.filestreaming.download_generation import build_data_file_name
+from usaspending_api.download.filestreaming.download_source import DownloadSource
+from usaspending_api.download.management.commands.delta_downloads.award_financial.federal_account import (
+    AccountDownloadDataFrameBuilder,
+    AccountDownloadFilter,
+)
+from usaspending_api.download.lookups import JOB_STATUS_DICT, FILE_FORMATS, VALUE_MAPPINGS
+from usaspending_api.download.models import DownloadJob
+from usaspending_api.download.v2.request_validations import AccountDownloadValidator, DownloadValidatorBase
+
+DOWNLOAD_SPEC = {
+    "award_financial": {
+        "federal_account": {
+            "df_builder": AccountDownloadDataFrameBuilder,
+            "validator_type": AccountDownloadValidator,
+        }
+    }
+}
+
+
+class Command(BaseCommand):
+
+    help = "Generate a download zip file based on the provided type and level."
+
+    download_job_id: int
+    download_job: DownloadJob
+    download_level: str
+    download_query: str
+    download_source: DownloadSource
+    download_spec: Dict
+    download_type: str
+    download_validator_type: Type[DownloadValidatorBase]
+    file_format_spec: Dict
+    file_prefix: str
+    jdbc_properties: Dict
+    jdbc_url: str
+    logger: Logger
+    should_cleanup: bool
+    spark: SparkSession
+    working_dir_path: Path
+
+    def add_arguments(self, parser):
+        parser.add_argument("--download-type", type=str, required=True, choices=list(DOWNLOAD_SPEC))
+        parser.add_argument(
+            "--download-level",
+            type=str,
+            required=True,
+            choices=set(
+                download_level
+                for download_level_list in [DOWNLOAD_SPEC[key] for key in DOWNLOAD_SPEC]
+                for download_level in download_level_list
+            ),
+        )
+        parser.add_argument("--download-job-id", type=int, required=True)
+        parser.add_argument("--file-format", type=str, required=False, choices=list(FILE_FORMATS), default="csv")
+        parser.add_argument("--file-prefix", type=str, required=False, default="")
+        parser.add_argument("--skip-local-cleanup", action="store_true")
+
+    def handle(self, *args, **options):
+        extra_conf = {
```

Comment:
If you happen to do any cleanup could you use the `DEFAULT_EXTRA_CONF` that defines these shared configs as constants?